deployment/src/main/java/io/quarkiverse/cxf/deployment/QuarkusCxfProcessor.java

```diff
@@ -212,93 +222,94 @@ public void build(
-        quarkusCapture c = new quarkusCapture(classOutput);
+        Bus bus = BusFactory.getDefaultBus();
+        // setup class capturing
+        bus.setExtension(new quarkusCapture(new GeneratedBeanGizmoAdaptor(generatedBeans)),
+                GeneratedClassClassLoaderCapture.class);
+
+        Set<String> clientSEIsInUse = findClientSEIsInUse(index);
 
         for (AnnotationInstance annotation : index.getAnnotations(WEBSERVICE_ANNOTATION)) {
             if (annotation.target().kind() != AnnotationTarget.Kind.CLASS) {
                 continue;
             }
 
             ClassInfo wsClassInfo = annotation.target().asClass();
+
             String sei = wsClassInfo.name().toString();
+
             reflectiveClass
                     .produce(new ReflectiveClassBuildItem(true, true, sei));
             unremovableBeans.produce(new UnremovableBeanBuildItem(
                     new UnremovableBeanBuildItem.BeanClassNameExclusion(sei)));
-            List<String> wrapperClassNames = new ArrayList<>();
 
             if (!Modifier.isInterface(wsClassInfo.flags())) {
                 continue;
             }
-            QuarkusJaxWsServiceFactoryBean jaxwsFac = new QuarkusJaxWsServiceFactoryBean();
-            Bus bus = BusFactory.getDefaultBus();
-            jaxwsFac.setBus(bus);
-            bus.setExtension(c, GeneratedClassClassLoaderCapture.class);
-            //TODO here add all class
-            try {
-                jaxwsFac.setServiceClass(Thread.currentThread().getContextClassLoader().loadClass(sei));
-                jaxwsFac.create();
-                wrapperClassNames.addAll(jaxwsFac.getWrappersClassNames());
-            } catch (ClassNotFoundException e) {
-                LOGGER.error("failed to load WS class : " + sei);
-            }
-            String pkg = wsClassInfo.name().toString();
-            int idx = pkg.lastIndexOf('.');
-            if (idx != -1 && idx < pkg.length() - 1) {
-                pkg = pkg.substring(0, idx);
-            }
-            AnnotationValue namespaceVal = annotation.value("targetNamespace");
-            String wsNamespace = namespaceVal != null ? namespaceVal.asString() : getNamespaceFromPackage(pkg);
-            String wsName = "";
-            if (annotation.value("serviceName") != null) {
-                wsName = annotation.value("serviceName").asString();
-            }
-            Collection<ClassInfo> implementors = index.getAllKnownImplementors(DotName.createSimple(sei));
+
+            // created on-demand if an implementor or client usage is found
+            QuarkusJaxWsServiceFactoryBean factoryBean = null;
+
+            String wsNamespaceDefault = Optional.ofNullable(annotation.value("targetNamespace"))
+                    .map(AnnotationValue::asString)
+                    .orElseGet(() -> getNamespaceFromPackage(wsClassInfo));
+            String wsNameDefault = Optional.ofNullable(annotation.value("serviceName"))
```

Comment:
same for wsNameDefault we can keep wsName to avoid new string for nothing.

Reply:
Same reason as for `wsNamespaceDefault`. I'll re-check this again.

Reply:
I think this must stay: `wsName` is calculated _per implementor_ (= per `wsClass`) and if that overwrites a single `wsName` that is set per outer loop (per `sei`), then it can leak into a client (when it doesn't use `@WebServiceClient`).

It's probably hard to see just here on GH. Better check it out locally.

But I just realized that `wsNameDefault` is never used for implementors; each implementor will always gate it's own `wsName`.
I'll split this up.

Reply:
done